joomla/joomla-cms
# [#33568] Remove mootools from com_content category view

### Issue

Today, the articles category view still loads MooTools, but it isn't needed anymore.
However the core.js JavaScript file is still needed for the sorting function within the default layout (the blog one doesn't has this).
### Solution

Currently, the layouts use `JHtml::_('behavior.framework);` which loads MooTools together with core.js. So far this was the only way to load core.js while using the Joomla API. With Joomla 3.3 we have now a new method  (`JHtml::_('behavior.core);`) to load core.js which will no longer load MooTools.

This PR makes removes the `behavior.framework` calls from the layouts and adds the `behavior.core` call to `JHtml::_('grid.sort')` which is the method that actually needs core.js.
### Testing Instructions

Create two menu items for an article category view. Once for the default layout and once for the blog one.
Make sure everything works as before while MooTools is no longer loaded.
If MooTools is still loaded after applying the patch, you may have to disable some modules or plugins because they may be loading MooTools.
**Important: This works only in Joomla 3.3!**
#### Tracker

http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=33568
